packages/mui-docs/src/Ad/AdProvider.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+
+export interface AdConfig {
+  /**
+   * The ratio of "ad display" event sent to Google Analytics.
+   * Used to avoid an exceed on the Google Analytics quotas.
+   * @default 0.1
```

Comment:
Why should this be configurable?

Reply:
I'm not really seeing the use case.

I have introduced this sampling because we didn't need to get all the events, we need enough to get statistical significance in the data. I started to collect this data to know the actual distribution of the ad display types. It's meant to make sure that reality matches the configuration intent.

I had surprises once I implemented this, there were bugs, the actual ad displayed types was different from what I tried to implement. It too me some time, but I eventually found the bugs. For this, it doesn't matter if the ad is shown in toolpad, base ui, etc.

Reply:
👍  All fine for me, I'm just stating a potential use-case. Personally I don't care for it neither.

Reply:
I don't see lot of usecases, so I'm ok to revert this. But IMO this migration to a dedicated package is an opportunity to find (and remove) all the hardcoded piece of code, and find clean way to make them configurable

Reply:
> all the hardcoded piece of code, and find clean way to make them configurable

@alexfauquette I agree with the strategy. In the tactics we use to do this, I think we should focus on looking at MUI X, Toolpad, Base UI docs in this order:

1. see what we had to duplicate because they couldn't configure it. Very clear signal.
2. see what we wish we were able to duplicate but couldn't, and this led to problematic UX. Clear signal.
3. see what we might need to configure in the future. Weak signal. Only worth it if there are no doubts?

This option seems to be case 3,